noise/conn.go

```diff
@@ -319,6 +320,9 @@ ContinueHandshake:
 			if !c.config.PublicKeyVerifier(hs.rs.PublicKey[:], receivedPayload) {
 				return errors.New("Noise: the received public key could not be authenticated")
 			}
+			c.isRemoteAuthenticated = true
+			c.remotePub = make([]byte, 32)
+			copy(c.remotePub, hs.rs.PublicKey[:])
```

Comment:
shouldn't you do this when receiving the `s` token?

Reply:
See comment above. I just want to get the static key when it has been verified. 
BTW, It is possible also to retrieve the value directly from the `PublicKeyVerifier` but having a `StaticKey` method is more practical code-wise.